src/libraries/Common/src/System/Net/Security/Unix/SafeDeleteNegoContext.cs

```diff
@@ -90,6 +102,12 @@ protected override void Dispose(bool disposing)
                     _acceptorCredential.Dispose();
                     _acceptorCredential = null;
                 }
+
+                if (_credential != null)
+                {
+                    _credential.DangerousRelease();
```

Comment:
This needs to be done even if disposing is false.  Otherwise, _credential will be forever leaked.